x-pack/docs/en/security/authentication/oidc-guide.asciidoc

```diff
@@ -0,0 +1,643 @@
+in {es}. See <<oidc-guide-authentication>>
+
+==== Service Account user for accessing the APIs
+
+The realm is designed with the assumption that there needs to be a privileged entity
+acting as an authentication proxy. In this case, the custom web application is the
+authentication proxy handling the authentication of end users ( more correctly,
+"delegating" the authentication to the OpenID Connect Provider ). The OpenID Connect
+APIs require authentication and the necessary authorization level for the authenticated
+user. For this reason, a Service Account user needs to be created and assigned a role
+that gives them the `manage_oidc` cluster privilege. The use of the `manage_token`
+cluster privilege will be necessary after the authentication takes place, so that the
+the user can maintain access or be subsequently logged out.
+
+[source,js]
+--------------------------------------------------
+POST /_security/role/facilitator-role
+{
+  "cluster" : ["manage_oidc", "manage_token"]
+}
+--------------------------------------------------
+// CONSOLE
+
+
+[source,js]
+--------------------------------------------------
+POST /_security/user/facilitator
+{
+  "password" : "<somePasswordHere>",
+  "roles"    : [ "facilitator-role"]
+}
+--------------------------------------------------
+// CONSOLE
+
+
+==== Handling the authentication flow
+
+On a high level, the custom web application would need to perform the following steps in order to
+authenticate a user with OpenID Connect:
+
+. Make an HTTP POST request to `_security/oidc/prepare`, authenticating as the `facilitator` user, using the name of the
+OpenID Connect realm in the {es} configuration in the request body. See the
+<<security-api-oidc-prepare-authentication,OIDC Prepare Authentication API>> for more details
++
+[source,js]
+--------------------------------------------------
+POST /_security/oidc/prepare
+{
+  "realm" : "oidc1"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[skip:These are properly tested in the OpenIDConnectIT suite]
++
+. Handle the response to `/_security/oidc/prepare`. The response from {es} will contain 3 parameters:
+  `redirect`, `state`, `nonce`. The custom web application would need to store the values for `state`
+  and `nonce` in the user's session (client side in a cookie or server side if session information is
+   persisted this way) and redirect the user's browser to the URL that will be contained in the
+  `redirect` value.
+. Handle a subsequent response from the OP. After the user is successfully authenticated with the
+  OpenID Connect Provider, they will be redirected back to the callback/redirect URI. Upon receiving
+  this HTTP GET request, the custom web app will need to make an HTTP POST request to
+  `_security/oidc/authenticate`, again - authenticating as the `facilitator` user passing the URL of
+  the GET request it received as a parameter, along with the values for `nonce` and `state` it had
```

Comment:
```suggestion
  the GET request it received as a parameter in the OP's response, along with the values for `nonce` and `state` it had
```
It's not 100% clear which URL you mean here, and my memory of OIDC is fuzzy. I assume the URL is from the OP.

Reply:
It is but it is the URL where it redirected the user and not a parameter, I will attempt to make this clear